stubs/docutils/docutils/statemachine.pyi

```diff
@@ -1,3 +1,194 @@
+    def get_source(self, line_offset: int) -> str: ...
+    def abs_line_offset(self) -> int: ...
+    def abs_line_number(self) -> int: ...
+    def get_source_and_line(self, lineno: int | None = None) -> tuple[str, int] | tuple[None, None]: ...
+    def insert_input(self, input_lines: list[str] | StringList, source: str) -> None: ...
+    def get_text_block(self, flush_left: bool = False) -> StringList: ...
+    def check_line(self, context: list[str], state: State, transitions: list[str] | None = ...) -> _TransitionResult: ...
+    def add_state(self, state_class: type[State]) -> None: ...
+    def add_states(self, state_classes: Sequence[type[State]]) -> None: ...
+    def runtime_init(self) -> None: ...
+    def error(self) -> None: ...
+    def attach_observer(self, observer) -> None: ...
+    def detach_observer(self, observer) -> None: ...
+    def notify_observers(self) -> None: ...
+
+class State:
+    patterns: dict[str, str | Pattern[str]] | None
+    initial_transitions: Sequence[str] | Sequence[tuple[str, str]] | None
+    nested_sm: type[StateMachine] | None
+    nested_sm_kwargs: dict[str, Any] | None
+    transition_order: list[str]
+    transitions: dict[str, tuple[Pattern[str], Callable[[], None], str]]
+    state_machine: StateMachine
+    debug: bool
+    def __init__(self, state_machine: StateMachine, debug: bool = False) -> None: ...
+    def runtime_init(self) -> None: ...
+    def unlink(self) -> None: ...
+    def add_initial_transitions(self) -> None: ...
+    def add_transitions(self, names: list[str], transitions) -> None: ...
+    def add_transition(self, name: str, transition: tuple[Pattern[str], str, str]) -> None: ...
+    def remove_transition(self, name: str) -> None: ...
+    def make_transition(self, name: str, next_state: str | None = None) -> tuple[Pattern[str], _TransitionMethod, str]: ...
+    def make_transitions(
+        self, name_list: list[str | tuple[str] | tuple[str, str]]
+    ) -> tuple[list[str], dict[str, tuple[Pattern[str], _TransitionMethod, str]]]: ...
+    def no_match(
+        self, context: list[str], transitions: tuple[list[str], dict[str, tuple[Pattern[str], _TransitionMethod, str]]]
+    ) -> _TransitionResult: ...
+    def bof(self, context: list[str]) -> tuple[list[str], list[str]]: ...
+    def eof(self, context: list[str]) -> list[str]: ...
+    def nop(self, match: Match[str], context: list[str], next_state: str) -> _TransitionResult: ...
+
+class StateMachineWS(StateMachine):
+    def get_indented(self, until_blank: bool = False, strip_indent: bool = True) -> tuple[StringList, int, int, bool]: ...
+    def get_known_indented(
+        self, indent: int, until_blank: bool = False, strip_indent: bool = True
+    ) -> tuple[list[str], int, bool]: ...
+    def get_first_known_indented(
+        self, indent: int, until_blank: bool = False, strip_indent: bool = True, strip_top: bool = True
+    ) -> tuple[list[str], int, int, bool]: ...
```

Comment:
Shouldn't the first return type in all of these be consistently either `list[str]` or `StringList`?